test/src/math/test_vecmat.cpp

```diff
@@ -0,0 +1,353 @@
+#include "math/staticrand.h"
+#include "math/vecmat.h"
+#include "util/FSTestFixture.h"
+
+#include <gtest/gtest.h>
+
+class VecmatTest : public test::FSTestFixture {
+  public:
+	VecmatTest() : test::FSTestFixture() { pushModDir("vecmat"); }
+
+  protected:
+	void SetUp() override { test::FSTestFixture::SetUp(); }
+	void TearDown() override { test::FSTestFixture::TearDown(); }
+};
+
+TEST_F(VecmatTest, test_vm_vec_add)
+{
+	for (size_t loop = 0; loop < 1000; ++loop) {
+		vec3d v1, v2, v3;
+
+		static_randvec(rand32(), &v1);
+		static_randvec(rand32(), &v2);
+
+		vm_vec_add(&v3, &v1, &v2);
+
+		for (size_t i = 0; i < 3; ++i) {
+			auto value1 = v1.a1d[i];
+			auto value2 = v2.a1d[i];
+			auto value3 = v3.a1d[i];
+
+			ASSERT_FLOAT_EQ(value3, value1 + value2);
+		}
+	}
+}
+
+TEST_F(VecmatTest, test_vm_vec_sub)
+{
+	for (size_t loop = 0; loop < 1000; ++loop) {
+		vec3d v1, v2, v3;
+
+		static_randvec(rand32(), &v1);
+		static_randvec(rand32(), &v2);
+
+		vm_vec_sub(&v3, &v1, &v2);
+
+		for (size_t i = 0; i < 3; ++i) {
+			auto value1 = v1.a1d[i];
+			auto value2 = v2.a1d[i];
+			auto value3 = v3.a1d[i];
+
+			ASSERT_FLOAT_EQ(value3, value1 - value2);
+		}
+	}
+}
+
+TEST_F(VecmatTest, test_vm_vec_add2)
+{
+	for (size_t loop = 0; loop < 1000; ++loop) {
+		vec3d v1, v2, v1backup;
+
+		static_randvec(rand32(), &v1);
+		static_randvec(rand32(), &v2);
+
+		vm_vec_copy_scale(&v1backup, &v1, 1.0f);
```

Comment:
Here (and in the function below) you make a backup by using `vm_vec_copy_scale()` with a scaling factor of `1.0f`, but the other functions (starting when testing `vm_vec_scale()` below), you assign one vector's `.xyz` property to another instead; I think that being consistent should be of primary importance.